### PR TITLE
feat: sentry - ADMT-491

### PR DESCRIPTION
**Pull request by marcelovicentegc**
This PR instruments the Admin environment (Admin and all Render Admin Apps) with Sentry.

For more details, get in touch with me on Slack.

### CODE REVIEW

File: react/utils/isAdmin.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      domain === 'myvtex.com' &&
+      (window.location.pathname.startsWith('/admin') ||
+        // Let's consider the Admin Login page as an Admin App even though it's technically not
+        window.location.pathname.startsWith('/_v/segment/admin-login/v1/login'))
+    ) {
+      return true
+    }
+  }
+
+  return false
```
Comment (lucasaarcoverde) [inline react/utils/isAdmin.ts:23]: just a suggestion to improve the readability

```suggestion
    const pathname = window.location.pathname
    const adminLoginPath = '/_v/segment/admin-login/v1/login'
    const isVtexDomain = domain === 'myvtex.com'
    const isAdminPath = pathname.startsWith('/admin') || pathname.startsWith(adminLoginPath)
    
    return isVtexDomain && isAdminPath
```

**Review by lucasaarcoverde — APPROVED** [review]
LGTM, just a suggestion

File: react/utils/isAdmin.ts (hunk 1 above)
Comment (marcelovicentegc) [reply react/utils/isAdmin.ts:23]: Good one @lucasaarcoverde, I'll apply it :hugs:

File: react/utils/isAdmin.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      '/admin',
+      '/_v/segment/admin-login/v1/login', // Let's consider the Admin Login page as an Admin App even though it's technically not.
+    ]
+
+    const { pathname } = window.location
+
+    return adminPathnames.some((path) => pathname.startsWith(path))
+  }
+
+  return false
```
Comment (matheusps) [inline react/utils/isAdmin.ts:27]:
```suggestion
  if (!canUseDOM) return false
  
  const { host } = window.location
  const domain = host.match(/(?<=\.)([^.]+\.[^.]+$)/)?.[0]

  if (domain !== 'myvtex.com') {
    return false
  }

  const adminPathnames = [
    '/admin',
    '/_v/segment/admin-login/v1/login', // Let's consider the Admin Login page as an Admin App even though it's technically not.
  ]

  const { pathname } = window.location

  return adminPathnames.some((path) => pathname.startsWith(path))
```

In find the early return is often better to read. Just a suggestion, apply if you think it looks better.

**Review by matheusps — APPROVED** [review]

File: react/error.tsx [hunk 3]
```diff
@@ -10,15 +11,27 @@ import ErrorImg from './images/error-img.png'
...
+      if (window.__ERROR__) {
+        captureException(window.__ERROR__)
+      } else if (window.__REQUEST_ID__) {
+        captureException(window.__REQUEST_ID__)
+      } else {
+        captureException(
+          'Render Runtime renderered an error page and there is no error or request id available'
+        )
+      }
+    }
```
Comment (vmourac-vtex) [inline react/error.tsx:34]: LGTM. 
Leaving a suggestion to improve readability:
 - invert `if` logic for early return
 - add variables for `__ERROR__` and `__REQUEST_ID__ `to minimize access (could also do it by destructuring assignment if you like)
 - fixed typo `renderered` in `defaultMessage`

```suggestion
  if (!isAdmin()) return;

  const error = window.__ERROR__;
  const requestId = window.__REQUEST_ID__;
  const defaultMessage =
    'Render Runtime rendered an error page, and there is no error or request ID available';

  if (error) {
    captureException(error);
  } else if (requestId) {
    captureException(requestId);
  } else {
    captureException(defaultMessage);
  }
 
```

**Review by vmourac-vtex — COMMENTED** [review]
LGTM, just submitted a suggestion

**Review by vmourac-vtex — APPROVED** [review]

File: react/utils/isAdmin.ts (hunk 2 above)
Comment (marcelovicentegc) [reply react/utils/isAdmin.ts]: Much better indeed @matheusps! Thanks. I'm applying it.

File: react/error.tsx (hunk 3 above)
Comment (marcelovicentegc) [reply react/error.tsx]: Much better @vmourac-vtex ! Thanks! I'm applying these changes.

**Pull request merged**